airflow/contrib/hooks/openfaas_hook.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.hooks.base_hook import BaseHook
+import requests
+from airflow import AirflowException
+
+OK_STATUS_CODE = 202
+
+
+class OpenFaasHook(BaseHook):
+    """
+    Interact with Openfaas to query, deploy, invoke and update function
+
+    :param function_name: Name of the function, Defaults to None
+    :type query: str
+    :param conn_id: openfass connection to use, Defaults to open_faas_default
+    :type conn_id: str
+    """
+
+    GET_FUNCTION = "/system/function/"
+    INVOKE_ASYNC_FUNCTION = "/async-function/"
+    DEPLOY_FUNCTION = "/system/functions"
+    UPDATE_FUNCTION = "/system/functions"
+
+    def __init__(self,
+                 function_name=None,
+                 conn_id='open_faas_default',
+                 *args, **kwargs):
+        self.function_name = function_name
+        self.conn_id = conn_id
+        super(BaseHook, self).__init__(*args, **kwargs)
+
+    def get_conn(self):
+        conn = self.get_connection(self.conn_id)
+        return conn
+
+    def deploy_function(self, overwrite_function_if_exist, body):
+        if overwrite_function_if_exist:
+            self.log.info("function already exist " + self.function_name + " going to update")
+            self.update_function(body)
+        else:
+            url = self.get_conn().host + self.DEPLOY_FUNCTION
```

Comment:
Need to add better handling, what are the required fields in the connection. We don't have any connection for `open_faas_default`.